services/server/src/handlers/report.ts

```diff
@@ -0,0 +1,140 @@
+  screenshot: boolean
+    .describe("filter by screenshot (uploaded or not)")
+    .optional(),
+  log: boolean.describe("filter by log files (uploaded or not)").optional(),
+  resolved: boolean.describe("filter by resolved value").optional(),
+  createdAt: dateFilter.describe("filter by creation time").optional(),
+  updatedAt: dateFilter.describe("filter by upate time").optional(),
+  page: pageNumber.optional().default(paginationDefaults.page),
+  size: pageSize.optional().default(paginationDefaults.size),
+});
+
+async function create(req: Request, res: Response, next: NextFunction) {
+  const user = req.user;
+  const allowed = isUser(user);
+  if (!allowed) return next(forbidden());
+
+  const { title, description }: IReport.CreateApiPayload =
+    createReportPayload.parse(req.body);
+
+  const key = uuid();
+
+  // upload the screenshot if exists
+  const screenshot = getRequestFile(
+    req.files,
+    IReport.AssetFileName.Screenshot
+  );
+  if (screenshot && exceedsSizeLimit(screenshot.size) !== false)
+    return next(largeFileSize());
+  const screenshotKey = screenshot
+    ? await upload({
+        data: screenshot.buffer,
+        type: screenshot.mimetype,
+        prefix: "reports/screenshots/",
+        key,
+      })
+    : undefined;
+
+  // upload the log file if exists
+  const log = getRequestFile(req.files, IReport.AssetFileName.Log);
+  if (log && exceedsSizeLimit(log.size) !== false) return next(largeFileSize());
+  const logKey = log
+    ? await upload({
+        data: log.buffer,
+        type: log.mimetype,
+        prefix: "reports/logs/",
+        key,
+      })
+    : undefined;
+
+  await reports.create({
+    userId: user.id,
+    title,
+    description,
+    screenshot: screenshotKey,
+    log: logKey,
+  });
+
+  res.sendStatus(200);
+}
+
+async function update(req: Request, res: Response, next: NextFunction) {
+  const user = req.user;
+  const allowed = isAdmin(user);
+  if (!allowed) return next(forbidden());
+
+  const { id, resolved }: IReport.UpdateApiPayload = updateReportPayload.parse(
+    req.body
+  );
+
+  const found = await reports.findById(id);
+  if (!found) return next(notfound.report());
+
+  await reports.update(id, { resolved });
+
+  res.sendStatus(200);
+}
+
+async function find(req: Request, res: Response, next: NextFunction) {
+  const user = req.user;
+  const allowed = isAdmin(user);
+  if (!allowed) return next(forbidden());
+
+  const payload: IReport.FindApiPayload = findReportsPayload.parse(req.body);
+
+  const { list, total } = await reports.find({
+    ids: payload.ids,
+    users: payload.users,
+    title: payload.title,
+    description: payload.description,
+    screenshot: payload.screenshot,
+    log: payload.log,
+    resolved: payload.resolved,
+    createdAt: payload.createdAt,
+    updatedAt: payload.updatedAt,
+    page: payload.page,
+    size: payload.size,
+  });
+
+  const response: IReport.FindApiResponse = { list, total };
```

Comment:
Please note that the items you are returing to the front end doesn't include the screenshot or the log file urls instead it only include its ids, we should generate a view link for these files before responding to the client. Check the function call `withImgUrl` and `withImgUrls` to get some idea.